envoy/api/v2/core/base.proto

```diff
@@ -39,6 +39,19 @@ message Percent {
   double value = 1 [(validate.rules).double = {gte: 0, lte: 100}];
 }
 
+// A truncated percentage is used in cases in which for performance reasons performing floating
+// point to integer conversions during randomness calculations is undesirable. The message includes
+// both a numerator and denominator that together determine the final truncated percentage in the
+// range 0-100 inclusive.
```

Comment:
Can you provide some examples of values and the resulting precision? It seems here we're expressing a general fractional value, rather than a percentage with some given precision. Will all code honor arbitrary precision that uses this? Will behavior always be truncate (floor function) vs. round?

Reply:
Yes can add more examples/explanation.